tests/core/factory/test_validation_definition_factory.py

```diff
@@ -634,6 +639,105 @@ def _test_overwrite_existing_validation(self, context: AbstractDataContext):
+        self._test_update_batch_definition(empty_cloud_context_fluent)
+
+    @pytest.mark.unit
+    @pytest.mark.xfail(
+        raises=ValidationError,
+        reason="Changing a BatchDefinition breaks loading a Validation Definition.",
+        strict=True,
+    )
+    def test_update_batch_definition__ephemeral(
+        self, ephemeral_context_with_defaults: EphemeralDataContext
+    ):
+        self._test_update_batch_definition(ephemeral_context_with_defaults)
+
+    def _test_update_batch_definition(self, context: AbstractDataContext):
+        # arrange
+        vd_name = self.random_name()
+        batch_def = self._build_batch_definition(context)
+        suite = context.suites.add(self._build_suite())
+        vd = ValidationDefinition(
+            name=vd_name,
+            data=batch_def,
+            suite=suite,
+        )
+        context.validation_definitions.add(validation=vd)
+
+        # act
+        new_batch_def_name = self.random_name()
+        batch_def.name = new_batch_def_name
+        new_vd = ValidationDefinition(
+            name=vd_name,
+            data=batch_def,
+            suite=suite,
+        )
+        updated_vd = context.validation_definitions.add_or_update(validation=new_vd)
+
+        # assert
+        assert updated_vd.data.name == new_batch_def_name
+        context.validation_definitions.get(vd_name)
+
+    @pytest.mark.filesystem
+    def test_replace_batch_definition__filesystem(self, empty_data_context: FileDataContext):
+        self._test_replace_batch_definition(empty_data_context)
+
+    @pytest.mark.cloud
+    def test_replace_batch_definition__cloud(self, empty_cloud_context_fluent: CloudDataContext):
+        self._test_replace_batch_definition(empty_cloud_context_fluent)
+
+    @pytest.mark.unit
+    def test_replace_batch_definition__ephemeral(
+        self, ephemeral_context_with_defaults: EphemeralDataContext
+    ):
+        self._test_replace_batch_definition(ephemeral_context_with_defaults)
+
+    def _test_replace_batch_definition(self, context: AbstractDataContext):
+        # arrange
+        vd_name = self.random_name()
+        batch_def = self._build_batch_definition(context)
+        suite = context.suites.add(self._build_suite())
+        vd = ValidationDefinition(
+            name=vd_name,
+            data=batch_def,
+            suite=suite,
+        )
+        existing_vd = context.validation_definitions.add(validation=vd)
+
+        # act
+        new_batch_def = self._build_batch_definition(context)
+        new_vd = ValidationDefinition(
+            name=vd_name,
+            data=new_batch_def,
+            suite=suite,
+        )
+        updated_vd = context.validation_definitions.add_or_update(validation=new_vd)
+
+        # assert
+        assert updated_vd.suite.id == existing_vd.suite.id
+        assert (
+            updated_vd.data.id != existing_vd.data.id
```

Comment:
I'd probably also assert that the id is non-None

Reply:
[9af4ef1](https://github.com/great-expectations/great_expectations/pull/10960/commits/9af4ef1fc2dd3f9c5d0754cf85e02aeff476df43)